consensus/blocknode.go

```diff
@@ -58,143 +41,103 @@ type blockNode struct {
 	delayedSiacoinOutputs map[SiacoinOutputID]SiacoinOutput
 }
 
-// CheckTarget returns true if the block id is lower than the target.
-func (b Block) CheckTarget(target Target) bool {
-	blockHash := b.ID()
-	return bytes.Compare(target[:], blockHash[:]) >= 0
+// childDepth returns the depth of a blockNode's child nodes. The depth is the
+// "sum" of the current depth and current target, where sum is defined as:
+//
+//     sum(x,y) := 1/(1/x + 1/y)
+func (bn *blockNode) childDepth() (depth Target) {
+	cumulativeDifficulty := new(big.Rat).Add(bn.target.Inverse(), bn.depth.Inverse())
+	return RatToTarget(new(big.Rat).Inv(cumulativeDifficulty))
 }
 
-// Int returns a Target as a big.Int.
-func (t Target) Int() *big.Int {
-	return new(big.Int).SetBytes(t[:])
+// heavierThan returns true if the blockNode is sufficiently heavier than
+// 'cmp', where "sufficient" is defined as:
+//
+//     (1/bn.depth) - (1/cmp.depth) > (1/cmp.target) * SurpassThreshold
+func (bn *blockNode) heavierThan(cmp *blockNode) bool {
```

Comment:
hmm. I guess it's a product of my FP instincts. Methods of the State can have side effects on the State. (In this case, the State is only being read, not modified -- but there's no way to know that without reading the function body.) In the new version, it's clear that only blockNodes are involved. Really, this function could take objects, not pointers, to make it even clearer that the nodes are not being modified.